### PR TITLE
[Chore](agg-state) add documentation about agg_state, add group_concat agg_state test case

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

add documentation about agg_state, add group_concat agg_state test case

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.56 seconds
 stream load tsv:          453 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          71.0 seconds inserted 10000000 Rows, about 140K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230625122743_clickbench_pr_167201.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.2 seconds
 stream load tsv:          456 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          61 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          68.9 seconds inserted 10000000 Rows, about 145K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230625123004_clickbench_pr_167212.html

File: docs/zh-CN/docs/sql-manual/sql-reference/Data-Types/AGG_STATE.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      properties("replication_num" = "1");
+
+插入数据示例：
+
+    insert into a_table values(1,max_by_state(3,1));
+
+查询数据示例：
+
+    select k1,max_by_merge(k2) from a_table group by k1 order by k1;
+
```
Comment (HappenLee) [inline docs/zh-CN/docs/sql-manual/sql-reference/Data-Types/AGG_STATE.md:58]: add same real case

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**